infer/src/base/Utils.ml

```diff
@@ -234,12 +249,21 @@ let with_file_out ?append file ~f =
   Exception.try_finally ~f ~finally
 
 
-let with_intermediate_temp_file_out file ~f =
+let with_intermediate_temp_file_out ?(retry = false) file ~f =
   let temp_filename, temp_oc = Filename.open_temp_file ~in_dir:(Filename.dirname file) "infer" "" in
   let f () = f temp_oc in
   let finally () =
     Out_channel.close temp_oc ;
-    Unix.rename ~src:temp_filename ~dst:file
+    let rec rename n =
+      try Unix.rename ~src:temp_filename ~dst:file
+      with e ->
+        if Int.equal n 0 then raise e
+        else
+          let delay = Random.float_range 0.1 0.3 in
+          ignore (Unix.nanosleep delay : float) ;
```

Comment:
should this do exponential backoff? anyhow, if it works it works

Reply:
Done anyway.